sdk/core/core-lro/src/lroEngine/locationPolling.ts

```diff
@@ -44,13 +43,39 @@ async function sendFinalRequest<TResult>(
   }
 }
 
-export function processLocationPollingOperationResult<TResult>(
+export function processLocationPollingOperationResult<
+  TResult,
+  TState extends PollOperationState<TResult>
+>(
   lro: LongRunningOperation<TResult>,
+  state: TState,
   resourceLocation?: string,
   lroResourceLocationConfig?: LroResourceLocationConfig
 ): (response: LroResponse<TResult>) => LroStatus<TResult> {
   return (response: LroResponse<TResult>): LroStatus<TResult> => {
-    if (isPollingDone(response.rawResponse)) {
+    const rawResponse = response.rawResponse;
+    const status = getStatus(rawResponse);
+    if (isUnexpectedPollingResponse(rawResponse) || status === "failed") {
+      throw new Error(`The long running operation has failed.`);
+    }
+    /**
+     * These HTTP request methods are typically used around provisioned resources
+     * so if the LRO was cancelled, there is nothing useful can be returned to
+     * the customer. POST requests on the other hand could support partial results
+     * so throwing an error in this case could be prevent customer from getting
+     * access to those partial results.
+     */
+    if (["PUT", "DELETE", "PATCH"].includes(lro.requestMethod) && status === "canceled") {
```

Comment:
If we don't throw even in PUT, DELETE and PATCH. Would users get stuck in a weird state or just wouldn't have interesting information in the request?

If they don't end up in a bad place, would it make sense to not throw and have everything fall into the next check?

Reply:
The `pollUntilDone()` method returns a result so if no error was thrown, it will return an object with a status field set to canceled as the result and it most likely will not match the expected type of the result object.

Reply:
I reverted the behavioral change for now and kept the original behavior of throwing in all cancellation scenarios. I will explore not throwing for POST requests in another PR.